guides/migrate-from-v2.md

```diff
@@ -0,0 +1,259 @@
+```
+
+If there is no such setter, we consider the property to be read-only.
+
+Most classes in OpenLayers 3 also inherit from the ol.Object class, which gives
+them a generic getter and setter for arbitrary properties. These can also be
+monitored for change events:
+
+```javascript
+var map = new ol.Map( /* config */ );
+map.on('change:humpty', function()({ /* handle the new value for humpty */ });
+map.set('humpty', 123);
+map.get('humpty');
+map.set('humpty', 456);
+```
+
+## Examples
+
+Let's have a look at some of the examples of the version 2 and see how we can
+write these as Openlayers 3 examples.
+
+### WMS layer
+
+The code behind the [OpenLayers 2 WMS
+example](http://dev.openlayers.org/examples/wms.html) mainly is:
+
+```javascript
+var lon = 5;
+var lat = 40;
+var zoom = 5;
+var map, layer;
+
+function init(){
+    map = new OpenLayers.Map( 'map' );
+    layer = new OpenLayers.Layer.WMS( "OpenLayers WMS",
+            "http://vmap0.tiles.osgeo.org/wms/vmap0", {layers: 'basic'} );
+    map.addLayer(layer);
+
+    map.setCenter(new OpenLayers.LonLat(lon, lat), zoom);
+}
+```
+
+Rewritten against the new API this becomes:
+
+```javascript
+var lon = 5;
+var lat = 40;
+var zoom = 5;
+var map, layer;
+
+function init(){
+  layer = new ol.layer.Tile({
+    source: new ol.source.TileWMS({
+      url: 'http://vmap0.tiles.osgeo.org/wms/vmap0',
+      params: { LAYERS: 'basic' }
+    })
+  });
+
+  map = new ol.Map({
+    layers: [layer],
+    target: 'map',
+    view: new ol.View({
+      center: [lon, lat],
+      zoom: zoom,
+      projection: 'EPSG:4326'
+    })
+  });
+}
+```
+
+The layer is configured to be a `ol.layer.Tile`, and we only specify a source
+for it. The `ol.source.TileWMS` only needs the `url` and some `params` to be
+able to create appropriate `GetMap`-urls. The `ol.Map` only has its `layers``
+set, defines a render-`target` and configures the `view`. 
+
+Even though the newer version is slightly longer, it has a clearer separation
+of concerns.
+
+### OSM layer
+
+The code behind the [OpenLayers 2 OSM
+example](http://dev.openlayers.org/examples/osm.html) mainly is:
+
+```javascript
+var map, layer;
+function init(){
+    map = new OpenLayers.Map( 'map' );
+    layer = new OpenLayers.Layer.OSM( "Simple OSM Map");
+    map.addLayer(layer);
+    map.setCenter(
+        new OpenLayers.LonLat(-71.147, 42.472).transform(
+            new OpenLayers.Projection("EPSG:4326"),
+            map.getProjectionObject()
+        ), 12
+    );    
+}
+```
+
+Rewritten against the new API this becomes:
+
+```javascript
+var map, layer;
+function init(){
+  var layer = new ol.layer.Tile({
+    source: new ol.source.OSM()
+  });
+  var map = new ol.Map({
+    layers: [layer],
+    target: 'map',
+    view: new ol.View({
+      center: ol.proj.transform([-71.147, 42.472], 'EPSG:4326', 'EPSG:3857'),
+      zoom: zoom
+    })
+  });
+}
+```
+
+In OpenLayers 2 we had a special layer for OSM, where in version three it is
```

Comment:
I guess we'll have to use versions consistently in this doc, now I see v3 or 3 or three etc.